src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -813,7 +818,14 @@ internal void RaiseDrop(global::Microsoft.UI.Xaml.DragEventArgs args)
 		{
 			if (_draggingOver?.Remove(args.SourceId) ?? false)
 			{
-				SafeRaiseEvent(DropEvent, args);
+				var args2 = new DragEventArgs(args);
+				SafeRaiseEvent(DragOverEvent, args2);
+				args.AcceptedOperation = args2.AcceptedOperation;
+				// On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire
+				if (args.AcceptedOperation is not DataPackageOperation.None)
+				{
+					SafeRaiseEvent(DropEvent, args);
+				}
```

Comment:
Same here:
```suggestion
				SafeRaiseEvent(DragOverEvent, args);
				
				// On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire
				if (args.AcceptedOperation is not DataPackageOperation.None)
				{
					args.Handled = false;
					SafeRaiseEvent(DropEvent, args);
				}
```

Reply:
Also _"On WinUI, if the final DragOver event sets AcceptedOperation to None, the DropEvent won't fire"_ : Is there any internal state that is clean in drop event handlers? If so we should make sure to clear it anyway.

Reply:
If I understand correctly, UIElementss should not expect to get a `DragDrop` every time. Cleaning up should instead happen in `DragCompleted` (which has a `Canceled` property), which we apparently don't actually raise.